### PR TITLE
Use parameter classes MergeParams + TransplantParams

### PR DESCRIPTION
**Pull request by snazy**
Also rename `CommitAttempts` to `CommitParams`.

This is just a refactoring, no change in functionality or public API.

### CODE REVIEW

File: versioned/persist/nontx/src/main/java/org/projectnessie/versioned/persist/nontx/NonTransactionalDatabaseAdapter.java [hunk 1]
```diff
@@ -234,17 +224,18 @@ public Hash merge(
...
-                    Collections.singletonList(from));
+                    Collections.singletonList(mergeParams.getMergeFromHash()));
 
             // Return hash of last commit (toHead) added to 'targetBranch' (via the casOpLoop)
-            return updateGlobalStatePointer(toBranch, pointer, toHead, newGlobalHead, newRefLog);
+            return updateGlobalStatePointer(
+                mergeParams.getToBranch(), pointer, newHead, newGlobalHead, newRefLog);
           },
-          () -> mergeConflictMessage("Retry-failure", from, toBranch, expectedHead));
+          () -> mergeConflictMessage("Retry-failure", mergeParams));
```
Comment (dimas-b) [inline versioned/persist/nontx/src/main/java/org/projectnessie/versioned/persist/nontx/NonTransactionalDatabaseAdapter.java:238]: tangential: should we perhaps report `currentHead` too?

**Review by dimas-b — APPROVED** [review]

File: versioned/persist/nontx/src/main/java/org/projectnessie/versioned/persist/nontx/NonTransactionalDatabaseAdapter.java (hunk 1 above)
Comment (snazy) [reply versioned/persist/nontx/src/main/java/org/projectnessie/versioned/persist/nontx/NonTransactionalDatabaseAdapter.java:238]: not here (this is "just" a retry) - but in general, yes: #3853

File: versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/CopyCommitsParams.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.adapter;
+
+import com.google.protobuf.ByteString;
+import java.util.function.Function;
+
+public interface CopyCommitsParams extends ToBranchParams {
```
[the remark this answers is not in the document]
Comment (snazy) [reply versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/CopyCommitsParams.java:21]: Technically not, but the inheritance comes from its say "natural use" (you need a "to-branch")

[the remark this answers is not in the document]
Comment (snazy) [reply versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/CopyCommitsParams.java:21]: > rename to MetadataRewriteParams?

Like that name!

**Review by dimas-b — APPROVED** [review]

**Pull request merged**